src/schema/user/user.model.ts

```diff
@@ -1,27 +1,74 @@
 // Libraries
-import {prop as Property, getModelForClass} from '@typegoose/typegoose';
-import {Field as GqlField, ObjectType as GqlObjectType} from 'type-graphql';
+import {Field, ObjectType, ID} from 'type-graphql';
+import {
+  prop as Property,
+  getModelForClass,
+  modelOptions,
+} from '@typegoose/typegoose';
+import {ObjectId} from 'mongodb';
 
-@GqlObjectType()
+// Models
+import {QuizModel, Quiz} from '../quiz/quiz.model';
+
+@modelOptions({options: {allowMixed: 0}})
+@ObjectType({description: 'The User Model'})
 export class User {
-  @GqlField()
-  readonly _id!: string;
+  @Field(() => ID, {description: 'User MongoDB ObjectID'})
+  id: ObjectId;
+
+  @Property({required: true, trim: true})
+  @Field({description: 'Name of the user'})
+  name: string;
 
-  @Property({default: new Date()})
-  @GqlField(() => Date)
-  createdAt: Date;
+  @Property({
+    required: true,
+    unique: true,
+    trim: true,
+    lowercase: true,
+    match:
+      /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@gmail(.([^<>()[\].,;:\s@"]{2,})){1,}$/i,
```

Comment:
@riteshsp2000 is only gmail allowed for user registration?

Reply:
are we planning to allow any others? if so ill update if not then we can keep it the same. And also I had google oauth in mind.